### DISCUSSION TITLE
Find a better name for Private/Public servers

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#2228] [Documentation]
First of all: THIS IS NOT ABOUT DISCUSSING THE IMPLEMENTATION OF REAL PRIVATE SERVERS. Please post suggestions about that in (https://github.com/jamulussoftware/jamulus/discussions/1057).

But now the main issue:

The term "private" and "public" server is outdated and technically incorrect. However the concept of a "private" server is still present and well known in the Jamulus community. We need to find a precise but also not too technical term for private servers.

@pljones suggested to change the term "private" server to "unregistered" server which I am not really happy with since I'd think it's not intuitive, tech speech and too much Jamulus related (ofc. there are also different arguments for "unregistered" ;-).)
I'd rather prefer something simpler and closer to "private" server like "Unpublished" (which is not ideal) or similar. 

_Also see https://github.com/jamulussoftware/jamulus/pull/2199#discussion_r776588256_

We're updating the documentation to use the term "unregistered" now: https://github.com/jamulussoftware/jamuluswebsite/pull/677

**Comment by gilgongo** [comment]
Possibly related https://github.com/jamulussoftware/jamulus/discussions/2100

**Comment by ghost** [comment]
Listed / Unlisted -- Like telephone numbers in the Telephone Directory (good old paper book).
Publicized / Unpublicized

**Reply by ann0see** [reply]
Yes. Actually I tend to like unlisted more then unregistered…

**Comment by dcorson-ticino-com** [comment]
Public sever / Independent server

**Comment by pljones** [comment]
As of right now (and subject to change at whim...), my working definitions are:

**Jamulus client** - connects your sound system to the internet!  Let's jam!

**Jamulus server** - somewhere Jamulus clients connect to for jams.  A Jamulus server has an "address".

**Server list** - a list of Jamulus servers retrieved from a Jamulus directory (a Jamulus directory "publishes" a server list, if you like to use the word "publish" for anything at all)

**Jamulus directory** - a service with which a Jamulus server can choose to _register_ and from which a Jamulus client can request a server list (list of Jamulus servers).  A Jamulus directory has an "address".

**Address** - Jamulus client users will most likely use the server list in the Connect dialog, so only need to know the Jamulus directory and Name for the Jamulus server they want to use.  However, the Conenct dialog also allows entering an Address.  An Address can also be given for a Custom directory, both in client and server.  The Address is a host name or IP (for a Jamulus server, both IPv4 and IPv6, when enabled) address with an optional port number, for instance "jams.example.com:22555" or "192.168.1.54".

**Public directory** - each of the built in Jamulus directories, along with some custom directories

**Custom directory** - additional Jamulus directories not built into the Jamulus server or client

**Private directory** - a custom directory that restricts access for server registration and server list retrieval

**Private server** - a server (optionally registered with a directory) that restricts access for client connection

NOTE: "_Privacy_" (i.e. running as a private directory or private server) is not directly supported by Jamulus - firewall set up is required.  (When thinking about this, remember `--showallservers` will return the full list of servers registered at a directory, regardless of whether the client can access them; hence saying the directory must also restrict client access.)

**Reply by gilgongo** [reply]
So the use of the word "private" is only correct when some network restriction is imposed, like firewalls or `--listfilter`? So the fact it's not registered doesn't mean it's private. Or the fact that the welcome message says it's private (in the case of a registered server).

**Reply by pljones** [reply]
> the fact it's not registered doesn't mean it's private

No, because I could post the address on Facebook ads and anyone could connect.  If I really didn't want anyone connecting except specific people, I'd need to ensure my firewall was set up right.  (Take Worldjam for example.  They firewall the Jamulus server instances where performances happen so no unexpected people join in.)

> or --listfilter

`--listfilter`...  Used where a Directory has a wider range of Clients than Servers, so the firewall has to let in to the Directory addresses that the Directory doesn't want registering.  (In the Worldjam example, their Directory could restrict registration just to the known servers whilst still remaining open to the world to view the registered servers.)

Taking that Worldjam Directory example further and combining with the Worldjam Server example -- when the Server firewalls become active, the Servers "disappear" from the server list... unless you use `--showallservers` - because the Servers stop responding to pings (except from clients on their firewall whitelist, that is).  And, because it shows all servers on a Directory, all the servers remain "visible" on explorer.jamulus.io, including their addresses.  But that doesn't mean you can connect to them, because the Server firewall stops the connect.

**Reply by gilgongo** [reply]
So in terms of the documentation, we will be retiring the use of  "private" in all contexts to avoid confusion. If anyone wants to write an article or something about "securing" your server/directory with firewalls and whatnot, then that's fine but we don't talk about it on the website. 

It just remains to decide if we want to adopt the terms "registered" and "unregistered" or "listed" and "unlisted" for this. The latter might be a little clearer perhaps since the former sort of implies something else happens once you're registered (like registering at the dentist, or being registered to vote).

**Reply by pljones** [reply]
I'm not that keen on the "un-" prefix, either, now I've had it pointed out to me.  You've either registered or you've not.

**Reply by gilgongo** [reply]
We can flex it, but I don't think we should impose assymetry for the sake of it like "The benefits of a registered server over one that isn't registered". Also read somewhere that more people skip "not" by accident when reading than they do an "un-" word because it looks different. Not sure how true that is, but it certainly happens in writing.

**Reply by pljones** [reply]
I certainly skip _writing_ "not" a lot... (when I intend to, that is).

**Comment by chrisrimple** [comment]
Mapping to @pljones' list, I like Listed/Unlisted or Registered/Unregistered.  I think they're the clearest terms and don't infer any specific sense of "privacy" (meaning that access is truly controlled at the server).

**Comment by dcorson-ticino-com** [comment]
I am wondering why a name change seems to be necessary.  I don't think that any of the proposals here, including mine, are better that the status quo.
Are we really seeing people having problems using and understanding Jamulus because of the server type naming?

**Reply by gilgongo** [reply]
> I don't think that any of the proposals here, including mine, are better that the status quo.

I tend to agree, but it's a bit like the debate about "Linux" vs "GNU/Linux" - it's hard to argue against precision once that issue is pointed out. On the one hand, to talk about public and private servers is good for new users because those terms are well understood outside of Jamulus, so it makes things easier. But on the other, we're being misleading about the detail.

I think you're right that we don't see people having problems because of the server type naming, and I would think that moving to less accurate but more precise terms will make Jamulus harder to understand. But it's a price worth paying because to carry on in the knowledge that the current vocabulary is "wong" will always be a problem I think. And once people understand what Jamulus means by "registered" or whatever, they will not be in danger of making incorrect assumptions.

**Reply by gilgongo** [reply]
See https://github.com/jamulussoftware/jamulus/discussions/2228#discussioncomment-1942612 for what I mean, @dcorson-ticino-com :-)

**Reply by dcorson-ticino-com** [reply]
Then I would suggest Listed server and Independent server.
I don't like using the negative prefix when not necessary.

**Reply by pljones** [reply]
Independent of what?  A server can be registered and "independent" of the Jamulus built in directories.  Definitely not a clear way of explaining it.

A server is registered if it's included in a directory's server list.

If a server is included in no directory's server list, it is not registered.

Simple English opposites.

**Reply by dcorson-ticino-com** [reply]
Then you have a suggestion what to call it that is positive, not using un- ?

**Reply by pljones** [reply]
Yes.  As stated in my message.

**Reply by pljones** [reply]
And I'll repost this, too:

----

Hm.  Might have to rescind my idea of
> Initially blank (i.e. if the user starts with "Directory" -> "None", it will remain blank)

The box holds a place holder of `STATUS` from the `.ui` layout, which gets overwritten with the current status -- i.e. "Not registered" if you're not registered - on start up, along with several other field updates (there's a method, `CServerDlg::UpdateGUIDependencies` that grabs current server values and stuffs them into the UI each time something changes - and on start up).

Currently, from start up:
```
$ ./Jamulus -s -i ini-server -p 56881
- server mode chosen
- initialization file name: ini-server
- selected port number: 56881
Using "192.168.1.19" as external IP.
Recording state not initialised
Recording state not initialised
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
```
![image](https://user-images.githubusercontent.com/1549463/148449276-bae5d970-c894-4c80-bb9f-cacbc3950703.png)
(That last paragraph is there in 3.8.1 -- I'm not sure anyone looking at "What's This?" text wants to know that...)
(Also... in the second paragraph, "select ... in" should be "select ... from".)
![image](https://user-images.githubusercontent.com/1549463/148446149-efdceaea-a977-42f3-8f31-62dc28bf22fb.png)
I decided the label for the Custom Directory address had to change and, as I was making related changes (i.e. "What's This?" text on the drop down), I should do it now...
![image](https://user-images.githubusercontent.com/1549463/148445995-60e25459-4d4c-4c99-b085-d86faca1e800.png)
Then switch to custom...
```
Server Registration Status update: Registration requested
```
![image](https://user-images.githubusercontent.com/1549463/148447942-d09ed8d8-01d6-4ad7-9730-3ae64e075afb.png)
... but that wasn't a directory:
```
Server Registration Status update: Registration failed
```
![image](https://user-images.githubusercontent.com/1549463/148448047-0f5fb595-3c0f-4353-8e75-01a788386b1a.png)
Switch back to "None" until I get the right Custom Directory address:
```
Server Registration Status update: Not registered
```
![image](https://user-images.githubusercontent.com/1549463/148448374-fe85499f-3acc-4752-bc0f-af0d4552c31b.png)
Cut-n-paste the right value...
![image](https://user-images.githubusercontent.com/1549463/148448666-7622de82-aeb3-431c-9cf8-d19295786a5c.png)
Switch to "Custom":
```
Server Registration Status update: Registration requested
Server Registration Status update: Registered
```
![image](https://user-images.githubusercontent.com/1549463/148448827-d7c3a505-58a7-4485-8536-f261b8f6a1f9.png)
(with the slightest flicker of "Registration requested" before succeeding).

Switch back to "None" now:
```
Server Registration Status update: Not registered
```
![image](https://user-images.githubusercontent.com/1549463/148449011-665467ec-9a84-41eb-812b-ac6dd1cf3ac1.png)

**Reply by dcorson-ticino-com** [reply]
> A server is registered if it's included in a directory's server list.
> If a server is included in no directory's server list, it is not registered.

After lots of thought and searching for other possibilities I still think the best solution is:
  A server is Listed if it's included in a directory's server list.
  If a server is included in no directory's server list, it is Independent.

There is otherwise no difference between the servers.  
Being a part of a directory server list (Listed) or being independent of any directory server list is the only difference.

**Reply by gilgongo** [reply]
I'm not sure if "independent" works very well with the fact that you can also connect directly ("independently" even) to a server that's listed on a directory. As @pljones says - independent of what?

**Reply by dcorson-ticino-com** [reply]
Independent of a directory server list.
You can connect directly to any server, that is not a differentiating characteristic.
The only difference is if a server is in a directory list or running independently from any directory list.

**Reply by pljones** [reply]
> You can connect directly to any server, that is not a differentiating characteristic.

That's not true.

**Reply by dcorson-ticino-com** [reply]
Why not?
If I know the URL or IP of any server I can at least try to connect.
Whitelisting is only available for directory servers.
The only difference I can think of is that a server listed on a directory server relieves the problem of port forwarding.

**Reply by ann0see** [reply]
You can't directly connect to a server registered to directory which is behind a NAT.

**Reply by dcorson-ticino-com** [reply]
> You can't directly connect to a server registered to directory which is behind a NAT.

Not sure how that rare technicality changes the argumentation.
So the directory does some translation telling the client what to say to get in touch.
If I know that translation I can connect directly, the data streams are not passing through the directory.
Or am I missing something?

**Reply by pljones** [reply]
> So the directory does some translation telling the client what to say to get in touch.

What does that mean?  Where is it in the code?

**Reply by dcorson-ticino-com** [reply]
I do not know, I am trying to make sense of what is said.
I am certainly missing something.

**Reply by gilgongo** [reply]
I can see what you mean, but in choosing "independent" as the opposite of "listed", you can see that "independent" falls down for similar reasons to "private" if it implies (or could be interpreted to mean) things that may not be true (like being "independent" of a Directory, which it might not be). The word "unlisted" (or "not listed") is free of those issues I think, and has the added advantage of being a linguistic pairing with "listed". The same is true of "registered" of course, which has a natural pairing, but (and perhaps this depends on the language) may not be as free of implications compared to "listed"?

**Comment by ann0see** [comment]
By reading through the comments, I have the impression that we‘re all kind of ok with calling previously public servers as "listed" servers. The problem is still the opposite.
Unlisted, (stand alone?), independent?

**Reply by gilgongo** [reply]
The advantage of "Not listed" or "Unlisted" (I think we could use both in the docs as needed) is that they linguistically pair well with "Listed".

**Reply by pljones** [reply]
If we're going to think of the directories like phone books, then you can be listed in the book... or not.  Take Yellow Pages as an example directory.  How many people here are listed in it?  I'd guess not many.  How do they think of themselves?  "Not listed in Yellow Pages" rather than something like "Independent of Yellow Pages"?

That you're not in _any_ directory still doesn't _mean_ you're not publicly, widely known - but you'll not be focussing on the usual target audience, admittedly.

**Reply by pljones** [reply]
OK, so next...

I'm not currently listed in Yellow Pages.  What do I need to do?  Personally, I don't think "I list" is the right answer to that.  I have to register -- a registration form to show I'm a business, so I can be accepted and then listed in their directory.

It's the directory that is providing the _listing_ to the clients.  The clients can then ask each server for more details (like "what's your ping, who's online?") -- the server is not "doing" the listing.  The server has to register (and re-register to say it's still a going concern).

**Comment by pgScorpio** [comment]
I really don't see the problem of the terms Private/Public.

All suggested alternatives like "unregistered", "unlisted", "independent", etc. are even more incorrect since even a private server can register to a private directory server, so it is still "registered" and "listed" and if it is only to be accessed via a private directory server it's not independent too,...

In my opinion "Public" means accessible by anyone via the public Jamulus explorer and "private" means you will have to have the URL or IP address provided by the owner.
Just like a public phone number is listed in the yellow pages and can be obtained by anyone, and a private phone number is to be obtained directly from the owner or one of his acquaintances...

**Reply by gilgongo** [reply]
Yes, hence the discussion :-) 

The counter-argument to yours is that "Public/Private" is not **strictly** correct if somebody thinks that in setting up a "private" server then Jamulus gives them some form of access control to it. 

I think having the Public/Private terminology helps people understand the differences between the two modes in a _general_ way (so it's accurate but not precise), and thereby makes it easier to understand what you see in the Jamulus UI without reading the documentation. But I also see the point that this may be misleading. 

So, if ease of understanding comes at the cost of being potentially misleading, it seems we need to choose a better term as a moral decision, if you like.

**Reply by pgScorpio** [reply]
> if somebody thinks that in setting up a "private" server then Jamulus gives them some form of access control to it

Well it does give them some access control as only people they give the address (and optional port number) to can access their server (just like with their private phone number).

I run a Jamulus helpdesk and I never had anyone who didn't understand this after just one or two lines of explanation ;=)
So maybe just add a line to the manual explaining Public/Private...
I really can't think of any naming that would be easier to explain.

**Reply by gilgongo** [reply]
> I really can't think of any naming that would be easier to explain.

Neither can I, and I totally agree with you (and BTW we do [explain it in the docs](https://jamulus.io/wiki/Running-a-Server#server-types)). The trouble is that the issue is less about explanation and more about accuracy. It's hard to argue against the fact that something isn't _objectively_ correct once that's pointed out. So in this case precision comes at the cost of explanation perhaps.

**Reply by pgScorpio** [reply]
> BTW we do [explain it in the docs](https://jamulus.io/wiki/Running-a-Server#server-types)

But this is in the server section, which most users don't read. ;=)

> So in this case precision comes at the cost of explanation perhaps.

I think clarity is more important than precision.... 
A lot of things in live are very clear while not precise (And precise things are often very unclear). On top of that I think everybody should know that nothing on the internet is really private. But that's my personal opinion,

**Reply by gilgongo** [reply]
> which most users don't read. ;=)

Heh - well we could link to that from the client docs maybe.

> I think clarity is more important than precision

That's my position, yes. But I also recognise the strength of feeling against that, which strength ultimately I think will cause more problems if it's ignored.

**Reply by pgScorpio** [reply]
I think we are mainly on the same level gilgongo, except that I don't recognize the strength of feeling against it. and I can't see how it would cause problems for anyone.

If we give in to this we should change a lot of terms since they are not precise...
Like: from the users view they are not a "Client" but more a "Participant" (Though from developers or server point of view Client is correct) so "New Client Level" should actually be "New Participant Level" ?
And is "Voice" "Steamer" or "Recorder" really an "Instrument"?
And is "Solo" still solo if you can set multiple participants to "Solo"?
And what is "Normal" audio quality? Also "High" and "Low" are not precise.

All are terms that are commonly used and very clear to most users, but they are absolutely not precise...

(I think I've made my point, so for now I will abandon this discussion. ;=) )

**Comment by mcfnord** [comment]
listed/unlisted

**Reply by pgScorpio** [reply]
@mcfnord

> listed/unlisted

But a private server can still be listed in a private directory server.... So private, but still listed... (And so also public to all having access to the private directory server.)

So I actually think there is nothing wrong with the terms Private/Public as long as documented clearly.

**Reply by mcfnord** [reply]
precision and accuracy are useful goals, but i want the word that best indicates what the operator might _do_ to achieve the result. AND I want to indicate what they _get_ when the do it. I need to register so i'll be... registered? or I need to add myself to the list, so i'll be listed. I don't think that means I'm public. It means I'm listed on a directory, some of which are well-known, public lists. Registering implies a middle step that we don't need to imply. We get listed on a directory, like a file that now appears when we list the directory files.

**Reply by pgScorpio** [reply]
>  I need to register so i'll be... registered?

Yes, I agree, "Registered" seems to be the correct term, though maybe not clear to the inexperienced user. But as I said: As long as documented clearly.... (no matter what term we use.)

EDIT: The terms "Private" and "Public" will remain also anyway, since you can register to a Private or a Public directory server.

**Reply by mcfnord** [reply]
What understanding does _registration_ add that _list_ lacks, for experienced users? I like killing words that add no understanding. And I like using the same and right word throughout a publication. I agree that the idea of "public directory" could work when describing our directories at well-known addresses.

**Reply by pgScorpio** [reply]
> What understanding does _registration_ add that _list_ lacks, for experienced users?

Well, it's all about what you do/did or what you (expect to) get.

If you register to a directory server that server will put you on it's list, so yes you should also be listed when registered, but that doesn't exclude that the directory server can do more than just listing, like logging connections etc. 
And who knows what the future will bring us? Maybe some time directory servers can even (temporarily) exclude specific servers from their list?
Also when the directory server is down you will still not be listed (though you are trying to register).

So in my opinion a normal server does "Register" and the directory server does "List".

It's all about point of view. And the documentation should cover all. Points of view will always differ from person to person and situation to situation anyway.

**Reply by mcfnord** [reply]
How does _register_ indicate what you get? Do most users care that a directory might log (and aren't we saying by default they do not?) Why hold on to a lesser word just because we might need it in the future? Why reduce clarity and simplicity for a corner case of a down directory? We're discussing documentation that _covers all_ in the sense that people making a decision about listing needs to understand _the basics_ needed to make the decision. This includes the language used. Even today, I helped a man register his new server. The word did not convey that he was now "listed" anywhere. A person lists their server on a directory. And since it's a decision every server operator needs to make, we need to present it in an intentionally simplified explanation that does not aspire to cover advanced or unusual scenarios.

**Reply by pgScorpio** [reply]
> How does _register_ indicate what you get? 

It doesn't, _list_ should as you stated ... 

_register_ is the action, _listed_ should be the result. (But as I made clear that's not guaranteed either)

> Do most users care that a directory might log (and aren't we saying by default they do not?) Why hold on to a lesser word just because we might need it in the future?
[...]
we need to present it in an intentionally simplified explanation that does not aspire to cover advanced or unusual scenarios.

No they probably don't, but might... 

And again, we need to document properly, this includes using all correct terminology and not excluding the possibility of future changes.
We should certainly not over-simplify by leaving out steps and/or using wrong terms.
Note that someone setting-up a server is not the average user and some technical insight (and reading the manual) should be expected.

====
@mcfnord 

As far as I can see the only thing that started our discussion was my remark that "registered" was the correct term (for normal servers) and that the terms "Private" and "Public" still remain valid also (for directory servers). I never said "listed" is a wrong term (though it is an 'expected' result, not an action) , and I just say we should document correctly, using ALL correct terms and action/result relations.

So I really don't understand why you insist on using "listed" instead of "registered" since these are two different things.

** As I vented my thougts on this subject I rest my case and leave this discussion again... **

**Reply by mcfnord** [reply]
The steps we require of a new server operator, the information we expose to them, and what we expect them to learn needs to be minimized and simplified. Otherwise, friction. Which reminds me of another mark against _registered_: We ask every server operator to register, even if they ultimately want an unregistered server. (It's a test step.) Typically, _registering_ means submitting personally identifiable information. But we don't do that. I bet some people stop when they realize they need to "register". It's a "high friction" word we should avoid. So if I had my way, someone setting up a server wouldn't see the word "register" even if more advanced topics mention it.  

As for _public_, it seems a little shaky. Listed on a public directory? Maybe that makes your server public. Private is a worse word, because really all that's going on is the server isn't listed on a public directory (or maybe lists on no directory). You can still call an unlisted number if you know the digits! So when is a server really private? But you're saying a directory server could be public or private. Same thing: Its address is well-known or unknown, but not public or private. 

I know if I continue using the elegant language that omits registration talk, and instead talk about listing a server on a directory, or running an unlisted server, the ease and clarity of this language will speak for itself.

**Reply by gilgongo** [reply]
> Typically, registering means submitting personally identifiable information.

Only if the operator chooses to supply such information when creating a registered server.

On the vocabulary: the issue here (now resolved) was that "public" and "private" were clearly misleading terms. Whether "registered" should be "listed" (or any other term) is a moot point because what matters more are the concepts. A registered server could be called a "banana server" but you'd still set it up the same. Also don't forget we are multi-lingual, so any stylistic discussion of English terms is pretty limited.

**Reply by mcfnord** [reply]
Where can I see page view counts across all languages? It's not moot if we're using two terms when one will do. Presumably we translate both terms.

**Reply by gilgongo** [reply]
We have no web analytics :-) What do you mean by "if we're using two terms when one will do"?

**Comment by mcfnord** [comment]
We're saying a server **lists** on a directory by **registering**. Those are the two terms. 

For non-English translations, presumably we compel new server operators to learn the translated form of registration, and then explain our vocabulary word just means adding your server to the directory list.

We should just say _a server lists itself on a directory_. [In fact, we do](https://jamulus.io/wiki/Running-a-Server). (We use _list_ so much! We understand that's what the step is about.) The registration process entails choosing a directory address and announcing yourself. This results in being listed. If you don't do it, your server is unlisted. Registration is an engineering concept that users don't need to know. By adding the vocabulary for every new server operator, we then have to educate about registration, when it's basically nothing more than listing, so far as I can see. Kill the term, win the simplicity of choosing a listed or unlisted server. "Join me on my unregistered server!" vs. "Join me on my unlisted server!"

**Reply by gilgongo** [reply]
When a Directory is full, a Server can be running as a "Registered server" without being listed by that Directory. This is the sort of issue we had with public/private, which were terms that led to misconceptions. Similarly, we have a distinction between term "Registered" and "listed". At least I think that's right @pljones ?

**Reply by pljones** [reply]
When a Directory is full, a registration request should be refused and the Server will not be registered.  When running with the UI this should be clear.  The logging when running headless should also be clear.

**Reply by gilgongo** [reply]
Yes, so I think that's why we use "Registered" rather than "Listed" as the Server configuration type/mode name. Since listed is not the same thing as registered (if you see what I mean).

**Reply by mcfnord** [reply]
I can't quite fathom what you mean. I think you're saying that since registration can fail, we don't want to call the server _listed_. The server just failed to list itself on the directory, because the directory is full. The result is an unlisted server that is configured to add itself to a directory list, but failed. This corner case can be explained in the log or UI. "Failed to list this server in the directory: The directory is full." And what's the result? An unlisted server. Technically, a server that failed to register, and so is not listed, but is called a registered server that failed to register when booted, a configuration failure corner case that we differentiate from a server that doesn't try to register, that is also unlisted as a result. If you run a server that lists on a directory, and that fails, it's a corner case that we can communicate in UI and log. I feel better able to discuss the scenarios, meanings, and outcomes more effectively without using the word _registered_. I should look at the UI and log in the failure scenario that makes this word necessary. The view that "a registered server could be called a banana server" also indicates to me that _registered_ isn't a great name, and that we're introducing an ambiguous term that we must then define. Without _registered_, we can use self-evident words like _add_ and _list_. If I fail to list on a directory, I have an unlisted server. How do I get an unlisted server if I want one? By not listing on a directory. I also might get one if that directory is full. I'll continue hoping to see the merit of _registered_.

**Reply by gilgongo** [reply]
OK but we just needed two separate terms: one for the type of Server and one for what happens when it appears on a Directory. The fact that we had to have a discussion to determine which words to use shows that there isn't an obvious choice. But like @DavidSavinkoff says, I don't think anyone has expressed any confusion about how to set up a server based on the vocabulary used today.

**Reply by pljones** [reply]
"To list" is an _active_ verb.  "To be listed" would be the passive.  Then there would be "a _request_ to be listed" which could fail.  The list entry contains some details - these are the "registration details".  The teminology would need to be something like "request a list entry containing listing details"?  (Because the "list", overall, contains "list entries", which are the details about the registered Server.)  So, that _request a list entry containing listing details_ is what "register" means and "registration" is the making of such a request.  I'm not sure using more words makes it clearer if the short forms are defined, clearly, once.

**Comment by ghost** [comment]
Lets make two lists of the same candidate words, one rated from best to worst, and the other rated from worst to best but not necessarily in reverse order. Then consider the first of the best and the last of the worst, and toss a coin for the rest.

**Reply by dcorson-ticino-com** [reply]
Whatever terms we decide on they will need to be described in the documentation.
I am not aware of any people having problems with the status quo.
Let's leave things as they are.
